From 0687378db4bfcf83b4a019c6c4fcbc1bab7a36e3 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:40:16 -0400
Subject: [virt] kvm/async_pf: kill the unnecessary use_mm/unuse_mm async_pf_execute()

Message-id: <1406040016-3289-222-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86300
O-Subject: [RHEL7 PATCH v2 221/221] KVM: async_pf: kill the unnecessary use_mm/unuse_mm async_pf_execute()
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Oleg Nesterov <oleg@redhat.com>

async_pf_execute() has no reasons to adopt apf->mm, gup(current, mm)
should work just fine even if current has another or NULL ->mm.

Recently kvm_async_page_present_sync() was added insedie the "use_mm"
section, but it seems that it doesn't need current->mm too.

Signed-off-by: Oleg Nesterov <oleg@redhat.com>
Reviewed-by: Andrea Arcangeli <aarcange@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit d72d946d0b649b79709b99b9d5cb7269fff8afaa)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

Conflicts:
 virt/kvm/async_pf.c [missing commit e0ead41a]
---

diff --git a/virt/kvm/async_pf.c b/virt/kvm/async_pf.c
index 7580aa1..a53dfa3 100644
--- a/virt/kvm/async_pf.c
+++ b/virt/kvm/async_pf.c
@@ -65,11 +65,9 @@ static void async_pf_execute(struct work_struct *work)
 
  might_sleep();
 
- use_mm(mm);
  down_read(&mm->mmap_sem);
  get_user_pages(NULL, mm, addr, 1, 1, 0, NULL, NULL);
  up_read(&mm->mmap_sem);
- unuse_mm(mm);
 
  spin_lock(&vcpu->async_pf.lock);
  list_add_tail(&apf->link, &vcpu->async_pf.done);
-- 
1.7.1